### PR TITLE
Introduce declared but not used inspection for literals (fixes #1310)

### LINKED ISSUE #1310: Function cannot be declared in another function body without being used
The continuation of #1308 
```go
func demo() {
    func(){} //func literal evaluated but not used
}
```

### PR DESCRIPTION
**Pull request by dlsniper**

### CODE REVIEW

File: src/com/goide/highlighting/GoAnnotator.java [hunk 1]
```diff
@@ -100,6 +100,33 @@ else if (element instanceof GoBuiltinCallExpr) {
         checkMakeCall(call, holder);
       }
     }
+    else if (element instanceof GoFunctionLit) {
+      checkEvaluatedButNotUsed(holder, element, "Function literal evaluated but not used");
+    }
+    else if (element instanceof GoStringLiteral) {
+      checkEvaluatedButNotUsed(holder, element, "String literal evaluated but not used");
```
Comment (ignatov) [inline src/com/goide/highlighting/GoAnnotator.java]: Do you have a spec for this inspection? Why do we check only literals?

File: src/com/goide/highlighting/GoAnnotator.java [hunk 2]
```diff
@@ -100,6 +100,33 @@ else if (element instanceof GoBuiltinCallExpr) {
...
+    }
+  }
+
+  private static void checkEvaluatedButNotUsed(AnnotationHolder holder, PsiElement element, String message) {
```
Comment (ignatov) [inline src/com/goide/highlighting/GoAnnotator.java]: Please annotate parameters.

**Comment by dlsniper** [comment]
I'll submit a different PR for further discussion, please revert this first.

**Comment by ignatov** [comment]
@mtoader Please don't merge PRs without review.

**Comment by mtoader** [comment]
@ignatov The things i cared about were resolved before I merged this.

**Pull request merged**